### PR TITLE
javascript/javascript2/week1

### PR DESCRIPTION
**Pull request by Rasim-Deniz**

### CODE REVIEW

File: javascript/javascript2/week1/homework.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    let shortestWord = words[0];
+    words.forEach(element => {
+        if (shortestWord.length > element.length) {
+            shortestWord = element;
+        }
+    });
+    return shortestWord;
+}
+console.log(findShortest(danishWords));
+
```
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js:15]: good work 💪, small comment, I would name the function findShortestWord

File: javascript/javascript2/week1/homework.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// Find and count the Danish letters
+
+function findDanishLetter(sentence) {
+    let æ = (sentence.match(/æ/gi) || []).length;
+    let ø = (sentence.match(/ø/gi) || []).length;
+    let å = (sentence.match(/å/gi) || []).length;
```
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js]: nice use of regex 👍

File: javascript/javascript2/week1/homework.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+};
+const danishString = "Jeg har en blå bil";
+findDanishLetter(danishString);  // returns {total: 1, å: 1}
+
+const danishString2 = "Blå grød med røde bær";
+findDanishLetter(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1}
+
+// Spirit animal name generator
+
+const spiritAnimals = ["Adventurous Wolf", "Mystical Unicorn", "Passionate Leopard", "Intuitive Dolphin", "Amazing Dove", "Victorious Elephant", "Magical Butterfly", "Wild Panther", "Creative Monkey", "Speedy Eagle"];
```
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js]: const spiritAnimals = [
  "Adventurous Wolf",
  "Mystical Unicorn",
  "Passionate Leopard",
  "Intuitive Dolphin",
  "Amazing Dove",
  "Victorious Elephant",
  "Magical Butterfly",
  "Wild Panther",
  "Creative Monkey",
  "Speedy Eagle",
];

In this format is easier to read the array
You can use VScode extension for formatting your code
for example Prettier - Code formatter

File: javascript/javascript2/week1/homework.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return;
+    }
+    else {
+        let userName = input.value;
+        let randomAnimal = spiritAnimals[Math.floor(Math.random() * spiritAnimals.length)];
+        let fullText = `${userName} - ${randomAnimal}`;
+        output.innerHTML = fullText;
+    }
+};
+
```
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js]: did you try to implement the mouseover event?

File: javascript/javascript2/week1/homework.js (hunk 1 above)
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js:15]: good work 👍  small comment, I would name the function findShortestWord

File: javascript/javascript2/week1/homework.js (hunk 2 above)
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js:21]: nice use of regex 👍

File: javascript/javascript2/week1/homework.js (hunk 4 above)
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js:62]: did you try to implement the mouseover event?

File: javascript/javascript2/week1/homework.js (hunk 3 above)
Comment (MirSzabo) [inline javascript/javascript2/week1/homework.js:44]: it is more readable this way
const spiritAnimals = [
  "Adventurous Wolf",
  "Mystical Unicorn",
  "Passionate Leopard",
  "Intuitive Dolphin",
  "Amazing Dove",
  "Victorious Elephant",
  "Magical Butterfly",
  "Wild Panther",
  "Creative Monkey",
  "Speedy Eagle",
];

use VS extension (for example Prettier) to format the code

**Review by MirSzabo — COMMENTED** [review]
nice homework 👍

File: javascript/javascript2/week1/homework.js (hunk 4 above)
Comment (Rasim-Deniz) [reply javascript/javascript2/week1/homework.js]: No

File: javascript/javascript2/week1/homework.js (hunk 3 above)
Comment (Rasim-Deniz) [reply javascript/javascript2/week1/homework.js]: Great suggestion, thank you.

File: javascript/javascript2/week1/homework.js (hunk 2 above)
Comment (Rasim-Deniz) [reply javascript/javascript2/week1/homework.js]: Thanks a lot

**Pull request merged**